tchannel-core/src/main/java/com/uber/tchannel/api/SubChannel.java

```diff
@@ -78,6 +78,12 @@ public SubChannel(String service, @NotNull TChannel topChannel) {
     }
 
     public SubChannel(String service, @NotNull TChannel topChannel, Connection.Direction preferredDirection) {
+        if (topChannel == null) {
```

Comment:
these can be checked by automated bytecode instrumentation, so there's no need for manual checks - `@NotNull` itself provides enough compile-time information.